### PR TITLE
Bug/url encode the right things

### PR DESCRIPTION
**Pull request by stefaneberl**

### CODE REVIEW

File: src/openkit-shared/Core/Util/PercentEncoder.cs [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+            var unreservedCharacters = BuildUnreservedCharacters(additionalReservedChars);
+            var resultBuilder = new StringBuilder(input.Length);
+
+            var index = 0;
+            while (index < input.Length)
+            {
+                var c = input[index];
+                if (IsUnreservedCharacter(c, unreservedCharacters))
+                {
+                    // unreserved character, which does need to be percent encoded
```
Comment (y0chanan) [inline src/openkit-shared/Core/Util/PercentEncoder.cs]: does not need

File: tests/openkit-shared-tests/Core/Configuration/OpenKitConfigurationTest.cs [hunk 2]
```diff
@@ -172,6 +175,25 @@ public void ADefaultConstructedConfigurationUsesStrictTrustManager()
...
+        {
+            // given
+            var target = CreateDefaultConfig();
+
+            // then
+            Assert.That(target.ApplicationID, Is.EqualTo("/App_ID%"));
+        }
+
+        [Test]
+        public void getApplicationIDPercentEncodedDoesPropperEncoding()
```
Comment (y0chanan) [inline tests/openkit-shared-tests/Core/Configuration/OpenKitConfigurationTest.cs]: test name starting with minor letter

shouldn't this be GetApplicationIDPercentEncodedDoesPropperEncoding

File: src/openkit-shared/Protocol/HTTPClient.cs [hunk 3]
```diff
@@ -39,10 +39,10 @@ public abstract class HTTPClient : IHTTPClient
         public class RequestType
         {
 
-            public static readonly RequestType STATUS = new RequestType("Status");              // status check
-            public static readonly RequestType BEACON = new RequestType("Beacon");              // beacon send
-            public static readonly RequestType TIMESYNC = new RequestType("TimeSync");          // time sync
-            public static readonly RequestType NEW_SESSION = new RequestType("NewSession");     // new session
+            public static readonly RequestType Status = new RequestType("Status");              // status check
```
Comment (thomasgrassauer) [inline src/openkit-shared/Protocol/HTTPClient.cs:42]: +1 for renaming

**Review by y0chanan — CHANGES_REQUESTED** [review]

File: src/openkit-shared/Core/Util/PercentEncoder.cs [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                    while (index < input.Length && !IsUnreservedCharacter(input[index], unreservedCharacters))
+                    {
+                        sb.Append(input[index]);
+                        index++;
+                    }
+                    // encode temp string using given encoding & percent encoding
+                    try
+                    {
+                        var encoded = encoding.GetBytes(sb.ToString());
+                        encoded.ForEach(b => resultBuilder.AppendFormat("%{0:X2}", b));
```
Comment (y0chanan) [inline src/openkit-shared/Core/Util/PercentEncoder.cs]: rename b to something more meaningful

**Review by thomasgrassauer — APPROVED** [review]

**Review by y0chanan — APPROVED** [review]

**Pull request merged**